baseapp/baseapp_test.go

```diff
@@ -2153,3 +2201,140 @@ func TestBaseApp_Init(t *testing.T) {
 		require.Equal(t, tc.expectedSnapshot.KeepRecent, tc.bapp.snapshotManager.GetKeepRecent())
 	}
 }
+
+func executeBlockWithArbitraryTxs(t *testing.T, app *BaseApp, numTransactions int, blockHeight int64) []txTest {
+	codec := codec.NewLegacyAmino()
+	registerTestCodec(codec)
+	r := rand.New(rand.NewSource(randSource))
+	randSource *= 2
```

Comment:
why?

Reply:
No particular reason. Modified it to an increment instead to avoid overflows.

Reply:
For clarity my question was about `randSource *=2`. Why do we want to modify it when we run this function?

Reply:
We'd like an `arbitrary` set of transactions, if we do not modify it, then the set of transactions will be the same each time as the `keyCounter` would start from the same point and the values would be the same